packages/frontend/src/app/create-store/page.tsx

```diff
@@ -132,14 +135,26 @@ const StoreCreation = () => {
               process.env.TEST ? undefined : new URL(window.location.href),
             );
             if (res.ok) {
+              setRelayClient(rc);
               const keyCardToEnroll = localStorage.getItem(
                 "keyCardToEnroll",
               ) as `0x${string}`;
               setIsMerchantView(true);
-              setRelayClient(rc);
               localStorage.setItem("merchantKeyCard", keyCardToEnroll);
-              setKeyCardEnrolled(true);
+              //Connect, authenticate, and send subscription request.
+              await rc.connect();
+              await rc.authenticate();
+              const filters = [
+                { objectType: ObjectType.OBJECT_TYPE_LISTING },
+                { objectType: ObjectType.OBJECT_TYPE_TAG },
+                { objectType: ObjectType.OBJECT_TYPE_ORDER },
+                { objectType: ObjectType.OBJECT_TYPE_ACCOUNT },
+                { objectType: ObjectType.OBJECT_TYPE_MANIFEST },
+                { objectType: ObjectType.OBJECT_TYPE_INVENTORY },
+              ];
+              await rc.sendSubscriptionRequest(shopId, filters);
```

Comment:
i'd probably add helper methods to relayClient, like `sendMerchantSubscriptionRequest` and `sendGuestSubscriptionRequest`.

In case we tweak the filter arguments, that's where i would look at least to make that change, not sprinkle it through the frontend.